core/src/main/java/org/apache/calcite/jdbc/CalciteMetaImpl.java

```diff
@@ -565,14 +565,22 @@ public MetaResultSet getTableTypes(ConnectionHandle ch) {
       }
       callback.execute();
       final MetaResultSet metaResultSet =
-          MetaResultSet.create(h.connectionId, h.id, false, signature, null);
+          new MetaResultSetImpl(h.connectionId, h.id, false, signature, null, updateCount);
```

Comment:
I would prefer updating Avatica to add another `MetaResultSet.create` that has `updateCount` parameter, instead of adding a subclass.

Reply:
This has been done in Avatica 1.16.0